[accountexeregister] iP

accountexeregister opened the pull request:
# CancelGPT

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))  

**CancelGPT** frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/accountexeregister/ip/releases/tag/v0.1).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉  

And it is `FREE`!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:  
```java
public class Main {
    public static void main(String[] args) {
        CancelGPT cancelGPT = new CancelGPT(Paths.get(System.getProperty("user.home"), "accountexeregister-ip", "data"));
        cancelGPT.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

accountexeregister committed:
Rename chatbot, add skeleton greeting

accountexeregister committed:
Add OOP, add echo

accountexeregister committed:
Add ability to store text in list and display list of texts stored

accountexeregister committed:
Add Task class, integrate Task in CancelGPT

accountexeregister committed:
Add mark and unmark task

accountexeregister committed:
Modify print output when mark/unmark task, displaying list

accountexeregister committed:
Add task subclasses, add support for tracking different types of tasks

accountexeregister committed:
Add automated text UI testing

accountexeregister committed:
Add error handling

accountexeregister committed:
Add support for deleting tasks

accountexeregister committed:
Refactor CancelGPT: remove unused methods, rename methods to more suitable names

accountexeregister committed:
Add command enums

accountexeregister committed:
Fix parsing task command creating Task objects with wrong description format

accountexeregister committed:
Update text-ui-test to include new commands and error handling

accountexeregister committed:
Add tasksStorage field to read and save tasks

accountexeregister committed:
Fix: add missing task storage reader

accountexeregister committed:
Add Task methods to create task object from saved data in tasks storage

accountexeregister committed:
Integrate reading and saving tasks to tasks storage file

accountexeregister committed:
Merge branch 'branch-Level-7'

accountexeregister committed:
Store dates in datetime

accountexeregister committed:
Merge branch 'branch-level-8'

accountexeregister committed:
Refactor: abstract save and read tasks to TasksStorage class

accountexeregister committed:
Refactor: abstract tasks operations to TasksList class

accountexeregister committed:
Refactor: abstract parsing and handling commands to CommandParser class

accountexeregister committed:
Refactor: abstract user interaction to UI class

accountexeregister committed:
Merge branch 'branch-A-MoreOOP'

accountexeregister committed:
Organize classes to packages

accountexeregister committed:
Merge branch 'branch-A-Packages'

accountexeregister committed:
Merge branch 'add-gradle-support'

Brendan8899 commented on src/main/java/CancelGPT/command/Command.java:
```diff
@@ -0,0 +1,23 @@
+package CancelGPT.command;
+
+public enum Command {
```
Good idea using Enums for Commands as there is a fixed set of Commands.
Upper case for all the names is a very good idea as it matches the Naming Conventions where constants should be all in Uppercase.

avrilmohh replied:
Missing javaDoc header comment

Brendan8899 reviewed (commented):
Look Great To Me!

avrilmohh commented on src/main/java/CancelGPT/task/Task.java:
```diff
@@ -0,0 +1,72 @@
+        return (isDone ? 1 : 0);
+    }
+
+    public static boolean getStatusBoolean(int statusInt) {
```
```suggestion
    public static boolean getStatus(int statusInt) {
```

avrilmohh commented on src/main/java/CancelGPT/core/CancelGPT.java:
```diff
@@ -0,0 +1,113 @@
+import java.util.List;
+import java.util.Scanner;
+
+public class CancelGPT {
```
Missing javaDoc header comment

avrilmohh reviewed (commented):
Good job! I really like that all your variable and method names are very clear on what their purpose is and also standardised their names e.g. "parseToDotaskCreationCommand" and "parseDeadlineTaskCreationCommand". This shows good naming code quality. However, while reading through your code, there were some parts I found hard to understand. Perhaps this could be prevented by adding JavaDoc comments? Other than that, excellent job. 😄

accountexeregister committed:
Modify CancelGPT and TasksStorage constructor to have storage directory path, add CancelGPT unit tests

accountexeregister committed:
Add unit test for ToDo

accountexeregister committed:
Merge branch 'branch-A-JUnit'

accountexeregister committed:
Configure build.gradle

accountexeregister committed:
Merge branch 'branch-A-Jar'

accountexeregister committed:
Add Javadoc

accountexeregister committed:
Tweak code to comply with Java coding standard

accountexeregister committed:
Add support for finding tasks by keyword

accountexeregister committed:
Merge branch 'branch-A-JavaDoc'

accountexeregister committed:
Merge branch 'branch-A-CodingStandard'

accountexeregister committed:
Merge branch 'branch-Level-9'

accountexeregister committed:
Add checkstyle

accountexeregister committed:
Fix checkstyle errors

accountexeregister committed:
Merge branch 'branch-A-CheckStyle'

accountexeregister committed:
Setup JavaFX

accountexeregister committed:
Create basic GUI

accountexeregister committed:
Make GUI echo user

accountexeregister committed:
Make CancelGpt echo on user input

accountexeregister committed:
Modify dialog boxes of CancelGpt and user in GUI to be on opposite sides

accountexeregister committed:
Refactor CancelGPT to use fxml

accountexeregister committed:
Complete GUI for CancelGpt

accountexeregister committed:
Refactor code

accountexeregister committed:
Merge branch 'branch-Level-10'

accountexeregister committed:
Add assertions

accountexeregister committed:
Improve code quality

accountexeregister committed:
Add FixedDurationTask class

Currently, there exists no class for an unscheduled task with a fixed
duration, which might be useful for users who want to create tasks
without a specific date in mind.

Let's create a class for this, named FixedDurationTask, which extends
from a Task to inherit its useful properties and methods. The duration
will be in the form of hours.

This is so that the data for this type of task can easily be saved and
read from.

accountexeregister committed:
Add handlers for FixedDurationTask

After adding FixedDurationTask class, we need to modify the codebase to
start integrating the FixedDurationTask so that users can manage this
type of task.

We do this by adding the appropriate commands, and methods, in a manner
similar to integrating other tasks.

We do this to make use of the extensibility of the codebase, and
maintain consistencyAdd handlers for FixedDurationTask

After adding FixedDurationTask class, we need to modify the codebase to
start integrating the FixedDurationTask so that users can manage this
type of task.

We do this by adding the appropriate commands, and methods, in a manner
similar to integrating other tasks.

We do this to make use of the extensibility of the codebase, and
maintain consistency.

accountexeregister committed:
Add assert for parsing fixed duration task creation command

accountexeregister committed:
Fix Javadoc

accountexeregister committed:
Merge pull request #2 from accountexeregister/branch-A-Assertions

branch-A-Assertions

accountexeregister committed:
Merge pull request #3 from accountexeregister/branch-A-CodeQuality

branch-A-CodeQuality

accountexeregister committed:
Merge pull request #4 from accountexeregister/branch-A-Streams

branch-A-Streams

accountexeregister committed:
Merge pull request #5 from accountexeregister/branch-B-FixedDurationTasks

branch-B-FixedDurationTasks

accountexeregister committed:
Add UI png

accountexeregister committed:
Merge branch 'branch-UI.png'

accountexeregister committed:
Add CancelGPT User Guide

accountexeregister committed:
Merge branch 'branch-A-UserGuide'

accountexeregister committed:
Rename Ui.png to UI.png in docs README

accountexeregister committed:
Merge branch 'branch-A-UserGuide'

accountexeregister committed:
Rename UI.png to Ui.png

accountexeregister committed:
Merge branch 'branch-A-UserGuide'
